src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs

```diff
@@ -156,16 +156,17 @@ private static IPAddress PrefixLengthToSubnetMask(byte prefixLength, AddressFami
             Debug.Assert((0 <= prefixLength) && (prefixLength <= 126));
             Debug.Assert((family == AddressFamily.InterNetwork) || (family == AddressFamily.InterNetworkV6));
 
-            byte[] addressBytes;
+            int byteCount;
             if (family == AddressFamily.InterNetwork)
             {
-                addressBytes = new byte[4];
+                byteCount = 4;
             }
             else
             {
                 Debug.Assert(family == AddressFamily.InterNetworkV6);
-                addressBytes = new byte[16];
+                byteCount = 16;
             }
+            Span<byte> addressBytes = stackalloc byte[byteCount];
```

Comment:
nit: Add a `Debug.Assert(byteCount == 16 || byteCount == 4)` **right above** the stackalloc line to highlight that byteCount is actually a small constant and not some unbounded variable length.

A comment instead would be fine too. This is mainly to make it easier for folks to review stackalloc usage, particularly if this code changes or where the value is set vs where its used (within stackalloc) move further apart.

Reply:
Added